tensorflow_addons/seq2seq/sampler.py

```diff
@@ -675,8 +693,14 @@ def sample(self, time, outputs, state):
 class InferenceSampler(Sampler):
     """A helper to use during inference with a custom sampling function."""
 
+    @typechecked
     def __init__(
-        self, sample_fn, sample_shape, sample_dtype, end_fn, next_inputs_fn=None
+        self,
+        sample_fn: Union[TensorLike, Callable],
+        sample_shape: tf.int32,
+        sample_dtype: tf.int32,
```

Comment:
TensorLike I think

Reply:
Thanks @gabrieldemarmiesse . I have tried without using `tf.int32`, most of time, I have an error msg, `TypeError: type of argument "sample_shape" must be one of (List, float, int, float16, float32, int8, int16, int32, int64, uint8, uint16, uint32, uint64, ndarray, Tensor); got tuple instead`, likewise for `sample_dtype`

Reply:
Sorry, I  meant for sample_shape.

Reply:
Thanks @gabrieldemarmiesse, not sure why having an error msg, `TypeError: type of argument "sample_shape" must be one of (List, float, int, float16, float32, int8, int16, int32, int64, uint8, uint16, uint32, uint64, ndarray, Tensor); got tuple instead`.  What do you think?

Reply:
I must have forgotten tuple in the list of compatible types for TensorLike https://github.com/tensorflow/addons/blob/3ddda238bc011132f37763649811159b77c9a0f4/tensorflow_addons/utils/types.py#L43

You can add tuple.